doc/source/server_architecture.rst

```diff
@@ -0,0 +1,86 @@
+
+Streams, Plugins and Handlers
+-----------------------------
+
+Data will generally flow through the components as follows:
+
+-> Inbound Stream -> (Handlers) -> Plugin -> Outbound Stream -> (Handlers) ->
+
+* Plugins can have any number of inbound streams as inputs and any number of outbound streams as outputs. 
+* Inbound streams can have either integer ports or other inbound stream names as inputs.
+* Outbound streams can have either plugin or inbound stream names as inputs. They can output to other outbound streams or to an integer port.
+* Streams can have any number of handlers that will be executed in sequence.
+
+In order to accomplish parallel processing paths with handlers, multiple streams that each contain a handler to be executed in parallel should be created and given the same input stream, so that when the original input stream receives a message it will pass it onto all streams subscribed to it, which will each execute their own handlers independently and concurrently.
+
+To customize the functionality of AIT, users may add custom handlers and plugins, and customize their configurations.
+
+All streams and plugins inherit from the base class :mod:`ait.core.server.client.ZMQClient` which handles the necessary ZeroMQ functionality. If you would like to add a plugin, it must inherit from :mod:`ait.core.server.plugin.Plugin` and implement the abstract `process` method which is called whenever the plugin receives a message from any of its inbound streams.
+
+There are several stream types that are instantiated based on the stream's configs. The AIT Server will check if the input or output types of streams are integers, and automatically instantiate the appropriate stream type.
+
+If you would like to create a custom handler, it must inherit from :mod:`ait.core.server.handler.Handler` and implement the `handle` method which is called whenever the stream it is attached to receives a message. 
+
+
+Configuring the server
+----------------------
+
+AIT uses `**config.yaml** <https://ait-core.readthedocs.io/en/master/configuration_intro.html>` to load configuration data for the server.
+
+Here is an example of the **server** portion of **config.yaml**:
+
+.. code-block:: none
+
+    server:
+        plugins:
+            - plugin:
+                name: ait.server.plugins.ait_gui_plugin
+                inputs: 
+                    - log_stream
+                    - telem_stream
+
+        inbound-streams:
+            - stream:
+                name: log_stream
+                input: 3077
+
+            - stream:
+                name: telem_stream
+                input: 3076
+                handlers:
+                    - name: ait.server.handlers.ait_packet_handler
+                      packet: 1553_HS_Packet
+
+        outbound-streams:
+            - stream:
+                name: command_stream
+                input: AitGuiPlugin
+                output: 3075
+
+Plugins
```

Comment:
Here's those sections I as talking about. I think including these configuration subsections with the stuff mentioned above would make sense